content/en/containers/amazon_ecs/apm.md

```diff
@@ -97,28 +98,34 @@ The [Amazon's ECS container metadata file][1] allows discovery of the private IP
 {{< code-block lang="curl" >}}
 cat $ECS_CONTAINER_METADATA_FILE | jq -r .HostPrivateIPv4Address
 {{< /code-block >}}
-
     
 [1]: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/container-metadata.html#metadata-file-format
 {{% /tab %}}
 {{< /tabs >}}
 
-Set the result as your Trace Agent hostname environment variable for each application container shipping to APM:
+Proivde the result of this request to the Tracer through the environment variable `DD_AGENT_HOST` for each application container sending APM traces.
 
-{{< code-block lang="curl" >}}
-os.environ['DD_AGENT_HOST'] = <EC2_PRIVATE_IP>
-{{< /code-block >}}
-    
-    
+### Set Tracer hostname
 
-## Launch time variables
+In cases where variables on your ECS application are set at launch time (Java, .NET, PHP), you **must** set the hostname as an environment variable with `DD_AGENT_HOST` using one of the above methods. If you have a startup script as your entrypoint this can be included in there, otherwise this can be added to the ECS Task Definition's `entryPoint`.
 
-In cases where variables on your ECS application are set at launch time, you **must** set the hostname as an environment variable with `DD_AGENT_HOST`. Otherwise, you can set the hostname in your application's source code for Python, Javascript, or Ruby. For Java and .NET, you can set the hostname in the ECS task. For example:
+For the other languages (Python, Javascript, Ruby, Go) you can alternatively set the hostname in your application's source code.
 
 {{< programming-lang-wrapper langs="python,nodeJS,ruby,go,java,.NET,PHP" >}}
 
 {{< programming-lang lang="python" >}}
 
+#### Launch time variable
+Update the Task Definition's `entryPoint` with:
+```json
+"entryPoint": [
+  "sh",
+  "-c",
+  "export DD_AGENT_HOST=$(curl http://169.254.169.254/latest/meta-data/local-ipv4); ddtrace-run python <your application>"
```

Comment:
Not all Python applications run with `ddtrace-run python <your application>"`, ie: uwsgi setups. What do you think of this?

```
"entryPoint": [
  "sh",
  "-c",
  "export DD_AGENT_HOST=$(curl http://169.254.169.254/latest/meta-data/local-ipv4); <command used to launch Python application container>"
```